core/models/fields/PrimaryKeyField.py

```diff
@@ -0,0 +1,35 @@
+from time import time
+from django.db.models import BigIntegerField
+import random
+
+START_TIME = 1537758068554
+
+
+def make_id():
+    """
+    Construct a BigInteger ID from the time and 23 random bits.
+
+    Only 1 in 8.3 million chance of getting the same result, even when two  objects are created at the exact same time
+    inspired by http://instagram-engineering.tumblr.com/post/10853187575/sharding-ids-at-instagram
+    """
+    t = int(time()*1000) - START_TIME
+    u = random.SystemRandom().getrandbits(23)
```

Comment:
I would use cryptographic-random instead. `Random.random` is only psuedo-random

Reply:
I mean... it doesn't need to be cryptographically random. Just random enough to not lead to collisions right?